src/research.py

```diff
@@ -88,89 +115,311 @@ def get_info(self):
+
+        #self.attack += random.randint(0, level)
+        self.defense += random.randint(0, level) * 2
+
+        #self.reload_time += random.randint(0, level)
+        #self.ammunition += random.randint(0, level) * 5
+        #self.projectile_speed += random.randint(0, level) 
+
+    def upgrade_skill(self, level):
+        #self.attack_skill_multiplier += random.randint(0, level)
+        self.defense_skill_multiplier += random.randint(0, level) / 5
+
     def get_defense(self, material):
         effective_defense = self.defense
 
         if material != None and self.material_multiplier > 0:
             effective_defense = int(effective_defense * self.material_multiplier * material.effect_strength)
 
-        return random.randint(0, effective_defense)
+        return random.randint(0, int(effective_defense))
 
     def copy(self):
-        return Armor(self.name, self.cost, self.material_multiplier, self.defense, self.defense_skill_multiplier)
+        return Armor(self.name, self.cost, self.material_multiplier, self.defense, self.defense_skill_multiplier, self.heavy)
 
     def __call__(self):
         return self.copy()
 
     def __repr__(self):
         return '{} (x{}): {} ({})'.format(self.name, self.material_multiplier, self.defense, self.defense_skill_multiplier)
 
+class Mount:
+    def __init__(self, name, cost, speed, attack, defense, attack_skill_multiplier, defense_skill_multiplier, heavy=False):
+        self.name = name
+
+        self.cost = cost
+
+        self.speed = speed
+        self.attack = attack
+        self.defense = defense
+        self.attack_skill_multiplier = attack_skill_multiplier
+        self.defense_skill_multiplier = defense_skill_multiplier
+
+        self.heavy = heavy
+
+    def get_info(self):
+        res = {}
+
+        res['name'] = self.name
+        res['cost'] = self.cost
+        res['speed'] = speed
+        res['attack'] = attack
+        res['defense'] = self.defense
+        res['attack_skill_multiplier'] = self.attack_skill_multiplier
+        res['defense_skill_multiplier'] = self.defense_skill_multiplier
+        res['heavy'] = heavy
+
+        return res
+
+
+    def get_defense(self):
+        effective_defense = self.defense
+        return random.randint(0, int(effective_defense))
+
+    def copy(self):
+        return Armor(self.name, self.cost, self.speed, self.attack, self.defense, self.attack_skill_multiplier, self.defense_skill_multiplier, self.heavy)
+
+    def __call__(self):
+        return self.copy()
+
+    def __repr__(self):
+        return '{} (x{}): {} ({})'.format(self.name, self.material_multiplier, self.defense)
+
+
 #------------------------------
 # WEAPON AND ARMOR DEFINITIONS
 #------------------------------
 
 #Short
-unarmed = Weapon('Unarmed', 0, 5, 0, 1, 1, 1, 1, reload_time=1)
-dagger = Weapon('Dagger', 10, 5, 1.5, 2, 2, 1.1, 1, reload_time=2)
+unarmed = Weapon('Unarmed', 0, 5, 0, 1, 1, 1, 1, reload_time=1, armor_pierce=-1)
```

Comment:
Does -1 armor piece mean that it can't pierce armor?

Not entirely clear on the mechanics of this.

Reply:
I might just redo the armor piercing thing.  It's not implemented the way I want it to be.